src/api/EscargotPublic.h

```diff
@@ -679,6 +679,44 @@ class ESCARGOT_EXPORT VMInstanceRef {
     Evaluator::EvaluatorResult executePendingJob();
 };
 
+class ESCARGOT_EXPORT DebuggerOperationsRef {
+public:
+    class WeakCodeRef;
+
+    struct BreakpointLocation {
+        BreakpointLocation(uint32_t line, uint32_t offset)
+            : line(line)
+            , offset(offset)
+        {
+        }
+
+        uint32_t line;
+        uint32_t offset;
+    };
+
+    typedef std::vector<BreakpointLocation> BreakpointLocationVector;
+
+    struct BreakpointLocationsInfo {
+        BreakpointLocationsInfo(WeakCodeRef* weakCodeRef)
+            : weakCodeRef(weakCodeRef)
+        {
+        }
+
+        // The codeRef is a unique id which is not garbage collected
+        // to avoid keeping script / function code in the memory.
+        WeakCodeRef* weakCodeRef;
+        BreakpointLocationVector breakpointLocations;
+    };
+
+    // Base class for debugger callbacks
+    class DebuggerClient {
+    public:
+        virtual void parseCompleted(StringRef* source, StringRef* srcName, std::vector<DebuggerOperationsRef::BreakpointLocationsInfo*>& breakpointLocationsVector) = 0;
```

Comment:
This could be changed to `std::vector<DebuggerOperationsRef::BreakpointLocationsInfo>*` so API user could take its ownership. The return value could be changed to an enum which tells if the ownership is taken.